### ISSUE TITLE
OmniPlan3.download's URL results in 404

### ISSUE THREAD

**Issue by nmcspadden** [autopkg/recipes#320]
**Describe the problem**
OmniPlan3.download goes to a URL that is a 404.

**AutoPkg output**
```
Processing OmniPlan3.download...
WARNING: OmniPlan3.download is missing trust info and FAIL_RECIPES_WITHOUT_TRUST_INFO is not set. Proceeding...
{'APPCAST_URL': 'https://update.omnigroup.com/appcast/com.omnigroup.OmniPlan3',
 'AUTOPKG_VERSION': '2.0.1',
 'CACHE_DIR': '/Users/Shared/AutoPkg/Cache',
 'FOO': 100,
 'JSS_IMPORTS': [{'name': 'example.com', 'password': 'assword'}],
 'MUNKI_REPO': '/Users/Shared/munki_repo',
 'NAME': 'OmniPlan3',
 'PARENT_RECIPES': ['/Users/Shared/AutoPkg/RecipeRepos/com.github.autopkg.recipes/OmniGroup/OmniGroupProduct.download.recipe'],
 'RECIPE_CACHE_DIR': '/Users/Shared/AutoPkg/Cache/com.github.autopkg.download.omniplan3',
 'RECIPE_DIR': '/Users/Shared/AutoPkg/RecipeRepos/com.github.autopkg.recipes/OmniGroup',
 'RECIPE_OVERRIDE_DIRS': ['/Users/Shared/AutoPkg/RecipeOverrides'],
 'RECIPE_PATH': '/Users/Shared/AutoPkg/RecipeRepos/com.github.autopkg.recipes/OmniGroup/OmniPlan3.download.recipe',
 'RECIPE_REPOS': {'/Users/Shared/AutoPkg/RecipeRepos/com.github.autopkg.recipes': {'URL': 'https://github.com/autopkg/recipes'}},
 'RECIPE_REPO_DIR': '/Users/Shared/AutoPkg/RecipeRepos',
 'RECIPE_SEARCH_DIRS': ['.',
                        '~/Library/AutoPkg/Recipes',
                        '/Library/AutoPkg/Recipes',
                        '/Users/Shared/AutoPkg/Recipes',
                        '/Users/Shared/AutoPkg/RecipeRepos/com.github.autopkg.recipes',
                        '/Users/Shared/AutoPkg/RecipeRepos/com.github.autopkg.recipes/OmniGroup'],
 'TEST_INT': 100,
 'verbose': 4}
SparkleUpdateInfoProvider
{'Input': {'alternate_xmlns_url': 'https://www.omnigroup.com/namespace/omniappcast/v1',
           'appcast_url': 'https://update.omnigroup.com/appcast/com.omnigroup.OmniPlan3'}}
SparkleUpdateInfoProvider: Version retrieved from appcast: 3.13.1
SparkleUpdateInfoProvider: Found URL https://downloads.omnigroup.com/software/MacOSX/10.14/OmniPlan-3.13.1.dmg
{'Output': {'url': 'https://downloads.omnigroup.com/software/MacOSX/10.14/OmniPlan-3.13.1.dmg',
            'version': '3.13.1'}}
URLDownloader
{'Input': {'url': 'https://downloads.omnigroup.com/software/MacOSX/10.14/OmniPlan-3.13.1.dmg'}}
URLDownloader: No value supplied for prefetch_filename, setting default value of: False
URLDownloader: No value supplied for CHECK_FILESIZE_ONLY, setting default value of: False
Traceback (most recent call last):
  File "/Users/nmcspadden/Applications/autopkg-autopkg/Code/autopkglib/__init__.py", line 673, in process
    self.env = processor.process()
  File "/Users/nmcspadden/Applications/autopkg-autopkg/Code/autopkglib/__init__.py", line 480, in process
    self.main()
  File "/Users/nmcspadden/Applications/autopkg-autopkg/Code/autopkglib/URLDownloader.py", line 327, in main
    raw_headers = super().download(curl_cmd)
  File "/Users/nmcspadden/Applications/autopkg-autopkg/Code/autopkglib/URLGetter.py", line 180, in download
    raise ProcessorError(f"curl failure: {curl_err} (exit code {retcode})")
autopkglib.ProcessorError: curl failure: The requested URL returned error: 404 Not Found (exit code 22)
  File "/Users/nmcspadden/Applications/autopkg-autopkg/Code/autopkglib/__init__.py", line 673, in process
    self.env = processor.process()
curl failure: The requested URL returned error: 404 Not Found (exit code 22)
Failed.
Receipt written to /Users/Shared/AutoPkg/Cache/com.github.autopkg.download.omniplan3/receipts/OmniPlan3-receipt-20191120-165431.plist

The following recipes failed:
    OmniPlan3.download
        Error in com.github.autopkg.download.omniplan3: Processor: URLDownloader: Error: curl failure: The requested URL returned error: 404 Not Found (exit code 22)

Nothing downloaded, packaged or imported.
```

**Comment by MichalMMac** [comment]
Looks like Sparkle feed was not updated to point to the latest file on their website. 
See https://downloads.omnigroup.com/software/MacOSX/10.14

**Comment by homebysix** [comment]
This appears to be resolved as of version 3.14.4:

```
% autopkg run -vv ~/Developer/_autopkg_org/recipes/OmniGroup/OmniPlan3.download.recipe
Processing ~/Developer/_autopkg_org/recipes/OmniGroup/OmniPlan3.download.recipe...
WARNING: ~/Developer/_autopkg_org/recipes/OmniGroup/OmniPlan3.download.recipe is missing trust info and FAIL_RECIPES_WITHOUT_TRUST_INFO is not set. Proceeding...
SparkleUpdateInfoProvider
{'Input': {'alternate_xmlns_url': 'https://www.omnigroup.com/namespace/omniappcast/v1',
           'appcast_url': 'https://update.omnigroup.com/appcast/com.omnigroup.OmniPlan3'}}
SparkleUpdateInfoProvider: Version retrieved from appcast: 3.14.4
SparkleUpdateInfoProvider: Found URL https://downloads.omnigroup.com/software/MacOSX/10.14/OmniPlan-3.14.4.dmg
{'Output': {'url': 'https://downloads.omnigroup.com/software/MacOSX/10.14/OmniPlan-3.14.4.dmg',
            'version': '3.14.4'}}
URLDownloader
{'Input': {'url': 'https://downloads.omnigroup.com/software/MacOSX/10.14/OmniPlan-3.14.4.dmg'}}
URLDownloader: No value supplied for prefetch_filename, setting default value of: False
URLDownloader: No value supplied for CHECK_FILESIZE_ONLY, setting default value of: False
URLDownloader: Storing new Last-Modified header: Thu, 01 Oct 2020 21:50:13 GMT
URLDownloader: Storing new ETag header: "d8224-4ba412d-5b0a300322f40"
URLDownloader: Downloaded ~/Library/AutoPkg/Cache/com.github.autopkg.download.omniplan3/downloads/OmniPlan-3.14.4.dmg
{'Output': {'download_changed': True,
            'etag': '"d8224-4ba412d-5b0a300322f40"',
            'last_modified': 'Thu, 01 Oct 2020 21:50:13 GMT',
            'pathname': '~/Library/AutoPkg/Cache/com.github.autopkg.download.omniplan3/downloads/OmniPlan-3.14.4.dmg',
            'url_downloader_summary_result': {'data': {'download_path': '~/Library/AutoPkg/Cache/com.github.autopkg.download.omniplan3/downloads/OmniPlan-3.14.4.dmg'},
                                              'summary_text': 'The following '
                                                              'new items were '
                                                              'downloaded:'}}}
EndOfCheckPhase
{'Input': {}}
{'Output': {}}
CodeSignatureVerifier
{'Input': {'input_path': '~/Library/AutoPkg/Cache/com.github.autopkg.download.omniplan3/downloads/OmniPlan-3.14.4.dmg/OmniPlan.app',
           'requirement': 'identifier "com.omnigroup.OmniPlan3" and anchor '
                          'apple generic and certificate '
                          '1[field.1.2.840.113635.100.6.2.6] /* exists */ and '
                          'certificate leaf[field.1.2.840.113635.100.6.1.13] '
                          '/* exists */ and certificate leaf[subject.OU] = '
                          '"34YW5XSRB7"'}}
CodeSignatureVerifier: Mounted disk image ~/Library/AutoPkg/Cache/com.github.autopkg.download.omniplan3/downloads/OmniPlan-3.14.4.dmg
CodeSignatureVerifier: Verifying code signature...
CodeSignatureVerifier: Deep verification enabled...
CodeSignatureVerifier: Strict verification not defined. Using codesign defaults...
CodeSignatureVerifier: /private/tmp/dmg.7q4Lz3/OmniPlan.app: valid on disk
CodeSignatureVerifier: /private/tmp/dmg.7q4Lz3/OmniPlan.app: satisfies its Designated Requirement
CodeSignatureVerifier: /private/tmp/dmg.7q4Lz3/OmniPlan.app: explicit requirement satisfied
CodeSignatureVerifier: Signature is valid
{'Output': {}}
Receipt written to ~/Library/AutoPkg/Cache/com.github.autopkg.download.omniplan3/receipts/OmniPlan3.download-receipt-20201017-154150.plist

The following new items were downloaded:
    Download Path                                                                                          
    -------------                                                                                          
    ~/Library/AutoPkg/Cache/com.github.autopkg.download.omniplan3/downloads/OmniPlan-3.14.4.dmg  
```
